### ISSUE TITLE
Allow filters on create-game page to restrict what button your opponent can choose

### ISSUE THREAD

**Issue by irilyth** [buttonmen-dev/buttonmen#1759]
If you create a game with "Any Button" as the opponent's button, and select filters on the game creation page, those filters should restrict what button your opponent can choose. (Whether your opponent is a specific opponent, or an open game against "Anybody".)

**Comment by jl8e** [comment]
The recent discussion on TL and filtering got me thinking about how best to implement this.

I see three approaches:

When a game/tournament is created, optionally attach a list of legal buttons.
Advantages: simple
Disadvantages: hard to display concisely, because there's no context. those lists take a relatively large amount of DB space, doesn't allow the list to change (for instance, if a TL button is added, a TL-only game or tournament wouldn't allow it

Allow boolean logic to be attached to a game/tournament:
(For instance: A TL button OR a non-TL without T&S)
Advantages: powerful, compact, not actually that bad to implement (though more complex than the first)
Disadvantages: complicated to display and allow players to interact with. Complicated to cleanly feed back to the API. The amount people actually want the power is likely pretty low.

Give a set of filter options, and string them together in a standardized way:
(For instance: (Set A OR Set B) AND (has speed OR rush)
Advantages: almost as powerful as boolean in practice, a bit more compact, about as easy to implement
Disadvantages: still awkward to display a complex filter
(It could also be done on top of full boolean on the back end, with access to the full power needing to go directly through the api, but I don't think it's worth it.)

Are there other approaches I'm overlooking?

(I have more thoughts on implementation and what types of filters we'd need, but I have to go make dinner right now.)

**Comment by blackshadowshade** [comment]
There's also the option of a list of legal buttons and/or sets.

**Comment by blackshadowshade** [comment]
Also, regarding database storage, the lists could be deleted once the game/tournament is complete, which is what I was intending to do.

**Comment by jl8e** [comment]
Yeah, post-completion deletion makes sense whatever method we use.

Things we might want to filter on:

Set (clearly an OR)
Not set (clearly an AND; can be faked by just ORing all the other sets)
Skill (can see this going either way, but OR makes more sense)
Not skill (AND)
Flags/not-flags (AND, I think)
Win % between X and Y
Win % within X of other button 
Not already in tournament

The last two are the ones that don't work with a static list.

Writing that list makes me less sure that we can get away without full boolean logic, but I think it's workable.
